### PR TITLE
Run unit tests in the CI

### PR DESCRIPTION
**Pull request by edbo**
* Fixes test compilation issue in token.rs
* Adds `cargo test` to CI
* Disables doc tests because they're not compiling

### CODE REVIEW

**Comment by edbo** [comment]
Successful test run here:

https://github.com/ShiftControl-io/rust/actions/runs/8581278439

**Comment by mrmauer** [comment]
Hey @edbo,
Thanks for the PR! The added CI task looks like a no brainer to me. The only thing here is that the testing fixes (ie ignoring doctests and adding the missing fields for a few tests) are all addressed now on main. Can you update your fork/branch first before we move forward?

**Comment by edbo** [comment]
@mrmauer Thanks for reviewing.

I've rebased against your main branch now so hopefully this is good to merge now.

File: src/propelauth/token.rs [hunk 1]
```diff
@@ -197,6 +197,7 @@ mod tests {
             first_name: Some("Easter".to_string()),
             last_name: Some("Egg".to_string()),
             username: None,
+            properties: None,
```
Comment (mrmauer) [inline src/propelauth/token.rs:200]: `properties` is also declared on line 204 below. Can you remove this and the other 2 instances of adding `properties` to a `User` below?

**Review by mrmauer — CHANGES_REQUESTED** [review]
Just need to remove redundant declarations of `properties` in the tests module.

**Comment by edbo** [comment]
@mrmauer Thanks! Sorry didn't spot that, running tests passes locally now.

**Review by mrmauer — APPROVED** [review]
LGTM! Thanks for the PR.

**Pull request merged**